src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -200,16 +214,17 @@ public static bool ExecuteAndVerifyEvents(FileSystemWatcher watcher, WatcherChan
         {
             bool result = true, verifyChanged = true, verifyCreated = true, verifyDeleted = true, verifyRenamed = true;
             AutoResetEvent changed = null, created = null, deleted = null, renamed = null;
-            string[] expectedFullPaths = expectedPaths == null ? null : expectedPaths.Select(e => Path.GetFullPath(e)).ToArray();
+            FileSystemEventHandler createHandler = null, changeHandler = null, deleteHandler = null;
```

Comment:
It would be better to keep the tuple and name it. Less likely to make a mistake or introduce one if you keep the related info together.

``` C#
bool verifyRenamed = (expectedEvents & WatcherChangeTypes.Renamed) > 0;
(AutoResetEvent EventOccured, RenamedEventHandler Handler) renamed = default;

if (verifyRenamed)
    renamed = WatchRenamed(watcher);

action();

if (verifyRenamed)
{
    bool actual = renamed.EventOccured.WaitOne();
    watcher.Renamed -= renamed.Handler;

    // Validate
}
```